apps/api/v1/pages/api/bookings/[id]/_auth-middleware.ts

```diff
@@ -30,40 +30,51 @@ async function authMiddleware(req: NextApiRequest) {
     }
   }
 
-  const userWithBookingsAndTeamIds = await prisma.user.findUnique({
+  const user = await prisma.user.findUnique({
     where: { id: userId },
-    include: {
-      bookings: true,
-      teams: {
-        select: {
-          teamId: true,
+    select: {
+      email: true,
+      bookings: {
+        where: {
+          id,
         },
       },
     },
   });
 
-  if (!userWithBookingsAndTeamIds) throw new HttpError({ statusCode: 404, message: "User not found" });
+  if (!user) throw new HttpError({ statusCode: 404, message: "User not found" });
 
-  const userBookingIds = userWithBookingsAndTeamIds.bookings.map((booking) => booking.id);
+  const userIsHost = user.bookings?.length;
 
-  if (!userBookingIds.includes(id)) {
-    const teamBookings = await prisma.booking.findUnique({
-      where: {
-        id: id,
-        eventType: {
-          team: {
-            id: {
-              in: userWithBookingsAndTeamIds.teams.map((team) => team.teamId),
-            },
+  const teamBookingsAsOwnerOrAdmin = await prisma.booking.findUnique({
+    where: {
+      id,
+      OR: [
+        {
+          attendees: { some: { email: user.email } },
+        },
+        {
+          eventType: {
+            OR: [
+              {
+                owner: { id: userId },
+              },
+              {
+                team: {
+                  members: {
+                    some: { userId, role: { in: ["ADMIN", "OWNER"] } },
+                  },
+                },
+              },
+            ],
           },
         },
-      },
-    });
+      ],
+    },
+  });
 
-    if (!teamBookings) {
-      throw new HttpError({ statusCode: 403, message: "You are not authorized" });
-    }
-  }
+  if (!userIsHost && !teamBookingsAsOwnerOrAdmin)
```

Comment:
The booking doesn't belong to the user as a host, or as a team-booking Owner or Admin. So they have no access to it.